runtime/reconcilers/project_parser.go

```diff
@@ -173,16 +174,25 @@ func (r *ProjectParserReconciler) Reconcile(ctx context.Context, n *runtimev1.Re
 	err = repo.Watch(ctx, func(events []drivers.WatchEvent) {
 		// Get changed paths that are not directories
 		changedPaths := make([]string, 0, len(events))
+		deletedPaths := make([]string, 0)
 		for _, e := range events {
 			if e.Dir {
 				continue
 			}
 			if parser.IsIgnored(e.Path) {
+				if !strings.HasPrefix(e.Path, "/tmp") && e.Type == runtimev1.FileEvent_FILE_EVENT_DELETE {
+					// only check for deleted folders that are not inside /tmp
+					deletedPaths = append(deletedPaths, parser.FindFilesInDir(e.Path)...)
+				}
```

Comment:
1. Skipping `/tmp` should be done separately at the start of the loop (might make sense to have a general exclude list global)
2. Why is the directory deletion case not handled under `if e.Dir { ...`?